src/frontend/src/common/queries/auth.ts

```diff
@@ -29,32 +24,28 @@ export const USE_USERDATA = {
   id: "userData",
 };
 
-export interface UserResponse extends APIResponse {
-  group: Group;
-  user: User;
-}
+const mapUserData = (obj: any): User => {
+  return <User>{
+    acknowledgedPolicyVersion: obj.acknowledged_policy_version,
+    agreedToTos: obj.agreed_to_tos,
+    group: mapGroupData(obj.group),
+    id: obj.id,
+    name: obj.name,
+    splitId: obj.split_id,
+  };
+};
 
-const USER_MAP = new Map<string, keyof User>([
-  ["auth0_user_id", "auth0UserId"],
-  ["group_admin", "groupAdmin"],
-  ["system_admin", "systemAdmin"],
-  ["group_id", "groupId"],
-  ["agreed_to_tos", "agreedToTos"],
-  ["acknowledged_policy_version", "acknowledgedPolicyVersion"],
-]);
+function mapGroupData(obj: any): Group {
+  return <Group>{
+    id: obj.id,
+    name: obj.name,
+  };
+}
 
-const fetchUserData = (): Promise<V2User> => {
+export const fetchUserInfo = (): Promise<User> => {
```

Comment:
I think all of this stuff (type de/serialization, error handling, etc etc) needs to be encapsulated in an API client library in the future, but for now, I'll add the intermediate interface just for clarity.

Reply:
I don't disagree about encapsulating it, but won't we still need to define that intermediate interface somewhere? (unless we can have a shared interface against the BE response schemas) We need to define somewhere what that initial fetched JSON looks like, right? ... Although, writing that out, maybe that deserializer step would handle it because it's going to parse the JSON and it either raises an error when it doesn't matched expected structure or it will match the expected structure, so the "raw" JSON interface aspect (what I called `FoobarRawResponse`) is implicit in the parser code?